src/core/response_cache.cc

```diff
@@ -0,0 +1,284 @@
+// Copyright (c) 2021, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions
+// are met:
+//  * Redistributions of source code must retain the above copyright
+//    notice, this list of conditions and the following disclaimer.
+//  * Redistributions in binary form must reproduce the above copyright
+//    notice, this list of conditions and the following disclaimer in the
+//    documentation and/or other materials provided with the distribution.
+//  * Neither the name of NVIDIA CORPORATION nor the names of its
+//    contributors may be used to endorse or promote products derived
+//    from this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS ``AS IS'' AND ANY
+// EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+// PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+// CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+// PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+// OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+// (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+#include "src/core/response_cache.h"
+
+namespace nvidia { namespace inferenceserver {
+
+RequestResponseCache::RequestResponseCache(const uint64_t size)
+{
+    // Allocate buffer
+    buffer_ = malloc(size);
+    // Exit early if buffer allocation failed
+    if (buffer_ == nullptr) {
+        throw std::runtime_error("failed to allocate buffer");
+    } 
+
+    // Create cache as managed buffer
+    managed_buffer_ = boost::interprocess::managed_external_buffer(
+        boost::interprocess::create_only_t{}, buffer_, size);
+    // TODO: Error check creation of managed_buffer_ ?
+}
+
+RequestResponseCache::~RequestResponseCache()
+{
+    // TODO: Free each managed buffer allocated to cache entries,
+    //       or is it enough to free full buffer?
+    for (auto& iter : cache_) {
+        auto& entry = iter.second;
+        for (auto& output : entry.outputs) {
+            managed_buffer_.deallocate(output.buffer);
+        }
+    }
+
+    // Free total cache buffer
+    if (buffer_ != nullptr) {
+        free(buffer_);
+    }
+}
+
+uint64_t RequestResponseCache::Hash(const InferenceRequest& request) {
+    std::size_t seed = 0;
+    // TODO: Linker ERROR
+    //boost::hash_combine(seed, request.ModelName()); 
```

Comment:
Similar linker issue here with `request.ModelName()` defined in `src/core/infer_request.cc` as opposed to `src/core/infer_request.h`